px4_ros2_cpp/include/px4_ros2/control/setpoint_types/fixedwing_lateral_longitudinal.hpp

```diff
@@ -0,0 +1,195 @@
+/****************************************************************************
+ * Copyright (c) 2025 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <px4_msgs/msg/fixed_wing_lateral_setpoint.hpp>
+#include <px4_msgs/msg/fixed_wing_longitudinal_setpoint.hpp>
+#include <px4_msgs/msg/lateral_control_configuration.hpp>
+#include <px4_msgs/msg/longitudinal_control_configuration.hpp>
+#include <Eigen/Eigen>
+#include <optional>
+
+#include <px4_ros2/common/setpoint_base.hpp>
+
+namespace px4_ros2
+{
+/** \ingroup setpoint_types_experimental
+ *  @{
+ */
+
+struct FwLateralLongitudinalSetpoint;
+struct FwControlConfiguration;
+
+/**
+ * @brief Setpoint type for fixedwing control
+*/
+class FwLateralLongitudinalSetpointType : public SetpointBase
+{
+public:
+  explicit FwLateralLongitudinalSetpointType(Context & context);
+
+  ~FwLateralLongitudinalSetpointType() override = default;
+
+  Configuration getConfiguration() override;
+
+  /**
+   * @brief Update the setpoint with full flexibility by passing a FwLateralLongitudinalSetpointType
+   * and FwControlConfiguration struct.
+   *
+   * @param setpoint a FwLateralLongitudinalSetpoint object where course, airspeed
+   * direction, lateral acceleration, altitude, height rate and equivalent airspeed
+   * can be set for full flexibility. See (## TODO: ADD LINK TO PX4_MSGS) for more information on this
+   * setpoint type.
+   *
+   * @param config a FwControlConfiguration object where pitch, throttle, lateral acceleration
+   * configuration can be set. See (## TODO: ADD LINK TO PX4_MSGS) for more information.
+   *
+   */
+
+  void update(
+    const FwLateralLongitudinalSetpoint & setpoint,
+    const FwControlConfiguration & config);
+
+  /**
+ * @brief Update the setpoint with full flexibility by passing a FwLateralLongitudinalSetpointType
+ *
+ * @param setpoint a FwLateralLongitudinalSetpoint object where course, airspeed
+ * direction, lateral acceleration, altitude, height rate and equivalent airspeed
+ * can be set for full flexibility. See (## TODO: ADD LINK TO PX4_MSGS) for more information on this
+ * setpoint type.
+ *
+ * @warning Any previously set configurations will be maintained when this method is called.
+ * If no configurations have been previously set, PX4 will set default configuration.
+ */
+
+  void update(const FwLateralLongitudinalSetpoint & setpoint);
+
+  /**
+* @brief Update the setpoint with simplest set of inputs. Lateral acceleration setpoints will be used as feedforward when course is provided.
+*  To directly control lateral acceleration, set course_sp to NAN. If a height rate is provided, the altitude setpoint
```

Comment:
So height rate isn't a feed-forward?
If so, should it not be split into 2 methods as they are mutually exclusive?
Like:
```
void updateWithAlt(
    const float altitude_amsl_sp, const float course_sp,
    std::optional<float> equivalent_airspeed_sp = {},
    std::optional<float> lateral_acceleration_sp = {});
void updateWithHeightRate(
    const float height_rate_sp, const float course_sp,
    std::optional<float> equivalent_airspeed_sp = {},
    std::optional<float> lateral_acceleration_sp = {});
```